R/estimate_ascertainment.R

```diff
@@ -0,0 +1,144 @@
+#'
+#' @param type A string, either `"static"` or `"varying"` which determines
+#' whether [cfr_static()] or [cfr_time_varying()] is used to calculate
+#' the resulting ascertainment ratio. Defaults to `"static"` if this argument is
+#' missing.
+#' @param severity_baseline A single number in the range 0.0 -- 1.0 for the
+#' assumed true baseline severity estimate used to estimate the overall
+#' ascertainment ratio. Missing by default, which causes the function to error;
+#' must be supplied by the user.
+#' @param max_date A `Date` representing a user supplied maximum date, up to
+#' which the time-varying severity estimate will be calculated. Useful in the
+#' case of long time-series, where the user wishes to focus on a specific
+#' time-period. See [as.Date()] for converting a string to a `Date`.
+#'
+#' @return A `<data.frame>` containing the maximum likelihood estimate estimate
+#' and 95% confidence interval of the corrected severity, named
+#' "ascertainment_mean" (for the central estimate), and "ascertainment_low" and
+#' "ascertainment_high" for the lower and upper interval limits.
+#' @export
+#'
+#' @examples
+#' # get data pre-loaded with the package
+#' data("covid_data")
+#' df_covid_uk <- covid_data[covid_data$country == "United Kingdom", ]
+#'
+#' df_covid_uk_subset <- subset(df_covid_uk, date <= "2020-05-31")
+#'
+#' # use a severity baseline of 1.4% (0.014) taken from Verity et al. (2020)
+#' # Lancet Infectious Diseases: 10.1016/S1473-3099(20)30243-7
+#'
+#' # use onset-to-death distribution from Linton et al. (2020)
+#' # J. Clinical Medicine: 10.3390/jcm9020538
+#'
+#' estimate_ascertainment(
+#'   data = df_covid_uk,
+#'   delay_density = function(x) dlnorm(x, meanlog = 2.577, sdlog = 0.440),
+#'   type = "varying",
+#'   severity_baseline = 0.014,
+#'   burn_in = 7L,
+#'   max_date = as.Date("2020-06-30")
+#' )
+#'
+estimate_ascertainment <- function(data,
+                                   delay_density = NULL,
+                                   type = c("static", "varying"),
+                                   severity_baseline,
+                                   burn_in = 7,
+                                   smoothing_window = NULL,
+                                   max_date = NULL) {
+  # input checking
+  # expect rows more than burn in value
+  checkmate::assert_data_frame(data, min.cols = 3, min.rows = burn_in + 1)
+  # check that input `<data.frame>` has columns date, cases, and deaths
+  checkmate::assert_names(
+    colnames(data),
+    must.include = c("date", "cases", "deaths")
+  )
+  # check for any NAs among data
+  checkmate::assert_data_frame(
+    data[, c("date", "cases", "deaths")],
+    any.missing = FALSE
+  )
+
+  checkmate::assert_number(
+    severity_baseline,
+    lower = 0.0, upper = 1.0, finite = TRUE
+  )
+  checkmate::assert_int(burn_in, lower = 0)
+  checkmate::assert_date(max_date, null.ok = TRUE)
+
+  # NOTE: delay_density is checked in estimate_outcomes() if passed and not NULL
```

Comment:
can be removed?

Reply:
We previously deceided to add this note to explain why the input checking was not done here, so will keep it.